### PR TITLE
Panoptes hookup - fetch Project data and Subjects for Classifier

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview
* In this PR, we're hooking up the custom front end with the Panoptes API.
* I'm focusing on the following implementations:
  * The CFE should be able to fetch project details from Panoptes
    * For example, staging project 1651, aka http://panoptes-staging-url/projects/darkeshard/transformers
  * The CFE should be able to fetch the queued subject for the current Zooniverse user looking at that specific project.
* See Issue #8 and Issue #10 for further spec details.
  * Specifically, look at [Task 1: Fetch Subjects from Panoptes](https://github.com/zooniverse/liberating-the-liberator/issues/8#issuecomment-307826733)
  * Also note [Sarah's comments](https://github.com/zooniverse/liberating-the-liberator/issues/8#issuecomment-307833120) on the subject queue endpoint.

Random thoughts:
* [Taxanomy? Ontology?] Right now the code's sitting in `src/ducks/classifier.js`, but I'm wondering if it should logically go into `src/ducks/projects.js` and `src/ducks/subjects.js`... hmmm.

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Actually, now that I think about it, I really should leave the `ClassifierContainer` component alone. I'll set up a new `ClassifierTester` component where we can more freely play around with the duck functionality and the Panoptes hooks without worrying about presentation.

**Comment by wgranger** [comment]
I think it makes more sense to separate these items into two ducks, `projects.js` and `subjects.js`. It'll be a lot more cleaner than potentially having a monolith `classifier.js`.

**Comment by shaunanoordin** [comment]
Update: I've created a simple `classifier-tester` that's viewable at http://localhost:3000/classify-tester . It does one very simple thing: fetch & display the details of staging Project 1651.

We can use this as the foundation for future Panoptes API data testing.

**Comment by shaunanoordin** [comment]
@wgranger I've added some extended comments for the Project duck and the Classify (Panoptes API) Tester; let me know if you have concerns about the code being too wordy/not wordy enough.

Ready for review!

**Comment by wgranger** [comment]
Travis hates that no environmental variables are set.

File: src/containers/ClassifierTester.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+class ClassifierTester extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <main className="app-content tester-page">
+        <div className="control-panel">
+          <button className="green button" onClick={this.fetchProject.bind(this)}>Fetch Project</button>
```
Comment (wgranger) [inline src/containers/ClassifierTester.jsx]: can this bind be put in the constructor? `this.fetchProject = this.fetchProject.bind(this)`

**Comment by wgranger** [comment]
Overall, this PR looks good and it ready to merge once the Travis issues and line comment are addressed.

**Comment by shaunanoordin** [comment]
Awright, I've added some updates:
* The event bind for `this.fetchProject()` in `ClassifierTester` has been moved to the constructor.
* Some updates to the Travis configuration have been made.

I don't think Travis has an issue with no environmental variables being set - I believe Travis is doing a default check using yarn (which isn't properly set up for our project) instead of our custom test script.

**Comment by shaunanoordin** [comment]
UPDATE: OK so the Travis problem may be affecting more people than just us.

1. Check the Slack conversation on #lab-app on 15 June 2017. Of note:
```
Sarah
[15:17] @here this may be the cause of the issues we're having with travis: https://kb.heroku.com/why-is-my-node-js-build-failing-because-of-conflicting-lock-files
[15:18] We should try to switch back to npm 5 managing packages, remove the yarn.lock file and see if travis will handle it
```

2. Also check out https://github.com/travis-ci/travis-ci/issues/7119 - it states how Travis tries to install yarn by default. (And apparently we have to... uh... overwrite npm install? Or something?)

True enough, if we check one of the job logs in this PR - e.g. https://travis-ci.org/zooniverse/liberating-the-liberator/jobs/242927829 - we'll see errors along the likes of `The command "eval yarn " failed. Retrying, 2 of 3.`

Heck if I know what's going on with that darn yarn.

**Comment by wgranger** [comment]
Seems like we still have some things to figure out with Travis. However, this shouldn't block this PR as taking the branch for a test run worked well. I'm going to merge, and we can figure out Travis separately.

**Comment by shaunanoordin** [comment]
Sweet - we'll open an investigation in Travis!
